WorkManagerSample/benchmark/src/androidTest/java/com/example/benchmark/WorkerBenchmark.kt

```diff
@@ -0,0 +1,123 @@
+    fun setUp() {
+        mContext = InstrumentationRegistry.getContext()
+        mTargetContext = InstrumentationRegistry.getTargetContext()
+    }
+
+    @Test
+    fun testBlurEffectFilterWorker() {
+        val worker = createWorker { appContext: Context, _: String,
+                                    workerParameters: WorkerParameters ->
+            BlurEffectFilterWorker(appContext, workerParameters)
+        }
+
+        benchmarkRule.measure {
+            val inputStream = inputStreamFor(mContext, JETPACK)
+            worker.applyFilter(BitmapFactory.decodeStream(inputStream))
+            inputStream?.close()
+        }
+    }
+
+    @Test
+    fun testGrayScaleFilterWorker() {
+        val worker = createWorker { appContext: Context, _: String,
+                                    workerParameters: WorkerParameters ->
+            GrayScaleFilterWorker(appContext, workerParameters)
+        }
+
+        benchmarkRule.measure {
+            val inputStream = inputStreamFor(mContext, JETPACK)
+            worker.applyFilter(BitmapFactory.decodeStream(inputStream))
+            inputStream?.close()
+        }
+    }
+
+    @Test
+    fun testWaterColorEffectFilterWorker() {
+        val worker = createWorker { appContext: Context, _: String,
+                                    workerParameters: WorkerParameters ->
+            WaterColorFilterWorker(appContext, workerParameters)
+        }
+
+        benchmarkRule.measure {
+            val inputStream = inputStreamFor(mContext, JETPACK)
+            worker.applyFilter(BitmapFactory.decodeStream(inputStream))
+            inputStream?.close()
+        }
+    }
+
+    private fun <T : ListenableWorker> createWorker(
+            createWorkerBlock: (appContext: Context, workerClassName: String,
+                                workerParameters: WorkerParameters) -> T
+    ): T {
+        lateinit var worker: T
+        val workerFactory = object : WorkerFactory() {
+            override fun createWorker(appContext: Context, workerClassName: String,
+                                      workerParameters: WorkerParameters): ListenableWorker? {
+                worker = createWorkerBlock(appContext, workerClassName, workerParameters)
+                return worker
+            }
+        }
+
+        val configuration = Configuration.Builder()
+                .setExecutor(SynchronousExecutor())
+                .setWorkerFactory(workerFactory)
+                .setMinimumLoggingLevel(Log.DEBUG)
+                .build()
+
+        WorkManagerTestInitHelper.initializeTestWorkManager(mTargetContext, configuration)
+        val workManager = WorkManager.getInstance()
+
+        // Enqueue a WorkRequest in a blocking way to guarantee WorkManager has instantiated the worker.
+        val workRequest = OneTimeWorkRequestBuilder<WaterColorFilterWorker>()
+                .setInputData(workDataOf(Constants.KEY_IMAGE_URI to JETPACK))
+                .build()
+        runBlocking {
```

Comment:
why runBlocking here?

Reply:
runBlocking is here so we can await the enqueue - although it's running on a synchronous executor we still have to await the promise as the API is async.

I'm open to some cleaner / better ideas to write this though as I don't like it much either :smiley: